sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -42,7 +42,144 @@ APIs for getting statistics about devices and modules, as well as service statis
 APIs for managing device identities, device twins, and querying devices
 
 ```csharp
-
+public class Devices
+{
+	/// <summary>
+	/// Create or update a device.
+	/// </summary>
+	/// <param name="deviceId">The unique identifier of the device to create.</param>
+	/// <param name="device">The device to create or update.</param>
+	/// <param name="ifMatch">A string representing a weak ETag for this device, as per RFC7232. The update operation is performed
+	/// only if this ETag matches the value maintained by the server, indicating that the device has not been modified since it was last retrieved.
+	/// To force an unconditional update, set If-Match to the wildcard character (*). For create operations, this value must be null.</param>
+	/// <param name="cancellationToken">The cancellation token.</param>
+	/// <returns>The created or updated device.</returns>
+	public virtual Response<DeviceIdentity> CreateOrUpdate(string deviceId, DeviceIdentity device, string ifMatch = null, CancellationToken cancellationToken = default)
```

Comment:
I can't find any Azure SDK recommendations for or against having a unified CreateOrUpdate function, so I'm just defaulting to how the service defined it. I'm very open to discussion on this one.

Reply:
This is technically CreateOrReplace right? Not sure if that word is standard to use. I thought in the discussion you had split this into Create, Replace and Update?

Reply:
Oh this is outdated :) My bad! Github confuses me.